pkg/cmd/roachtest/tests/cluster_to_cluster.go

```diff
@@ -196,6 +266,103 @@ func registerClusterToCluster(r registry.Registry) {
 			lv.assertValid(t)
 		},
 	})
+
+	for _, sp := range []replicationTestSpec{
+		{
+			srcKVNodes:       4,
+			dstKVNodes:       4,
+			cpus:             8,
+			pdSize:           1000,
+			workload:         replicateTPCC{warehouses: 1000},
+			timeout:          3 * time.Hour,
+			workloadDuration: 10 * time.Minute,
+			cutover:          5 * time.Minute,
+		},
+	} {
+		name := fmt.Sprintf("c2c/%s/cutover=%d/nodes=%dto%d/cpus=%d",
+			sp.workload.name(),
+			int(sp.cutover.Minutes()),
+			sp.srcKVNodes,
+			sp.dstKVNodes,
+			sp.cpus)
+
+		clusterOps := make([]spec.Option, 0)
+		clusterOps = append(clusterOps, spec.CPU(sp.cpus))
+		if sp.pdSize != 0 {
+			clusterOps = append(clusterOps, spec.VolumeSize(sp.pdSize))
+		}
+
+		r.Add(registry.TestSpec{
+			Name:            name,
+			Owner:           registry.OwnerDisasterRecovery,
+			Cluster:         r.MakeClusterSpec(sp.dstKVNodes+sp.srcKVNodes+1, clusterOps...),
+			RequiresLicense: true,
+			Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+
+				setup, cleanup := setupC2C(ctx, t, c, sp.srcKVNodes, sp.dstKVNodes)
+				defer cleanup()
+				m := c.NewMonitor(ctx, setup.src.kvNodes.Merge(setup.dst.kvNodes))
+
+				t.Status("populating source cluster before replication")
+				if initCmd := sp.workload.sourceInitCmd(setup.src.tenant.secureURL()); initCmd != "" {
+					c.Run(ctx, c.Node(setup.src.sqlNode), initCmd)
+				}
+
+				t.Status("starting replication stream")
+				streamReplStmt := fmt.Sprintf("CREATE TENANT %q FROM REPLICATION OF %q ON '%s'",
+					setup.dst.name, setup.src.name, setup.src.pgURL)
+				var ingestionJobID, streamProducerJobID int
+				setup.dst.sql.QueryRow(t, streamReplStmt).Scan(&ingestionJobID, &streamProducerJobID)
+
+				// begin the source cluster workload
+				workloadDoneCh := make(chan struct{})
+				m.Go(func(ctx context.Context) error {
+					defer close(workloadDoneCh)
+					cmd := sp.workload.sourceRunCmd(setup.src.tenant.secureURL(), sp.workloadDuration)
+					c.Run(ctx, c.Node(setup.src.sqlNode), cmd)
+					return nil
```

Comment:
Interestingly, if `c.Run` fails here, the test continues. I don't quite understand why yet.